haskell/cabal
# `package-constraints` by package file

Solves #8912 #9477

# Manual QA

By adding a field `package-constraints` in a cabal file and a list of dependencies with version ranges, for each of those dependencies, any dependency declaration that was unbounded will be set to the one on `default-constraints`:

```
package-constraints:
  x > 5

library a 
  build-depends: x

library b 
  build-depends: x >5.5

library c
  build-depends: y
```

Should be understood as:

```
library a 
  build-depends: x >5

library b 
  build-depends: x >5.5 && >5

library c
  build-depends: y
```

If the stanza is omitted, then nothing changes wrt previous version.

Notice this should not influence:
- `build-tool-depends`
- `pkgconfig-depends`
- transitive dependencies

--------------------------------------------------
